freqtrade/optimize/hyperopt_loss_calmar.py

```diff
@@ -0,0 +1,76 @@
+from datetime import datetime
+from pandas import DataFrame, Series
+from freqtrade.optimize.hyperopt import IHyperOptLoss
+import numpy as np
+from scipy.stats import norm
+
+NB_SIMULATIONS = 1000
+SIMULATION_YEAR_DURATION = 3
+HIGH_NUMBER = 100000
+CALMAR_LOSS_WEIGHT = 1
+SLIPPAGE_PERCENT = 0.001
+
+
+class CalmarHyperOptLoss(IHyperOptLoss):
+    """
+    Defines the default loss function for hyperopt
+    This is intended to give you some inspiration for your own loss function.
+    The Function needs to return a number (float) - which becomes for better backtest results.
+    """
+
+    @classmethod
+    def hyperopt_loss_function(cls, results: DataFrame, trade_count: int,
+                               min_date: datetime, max_date: datetime,
+                               *args, **kwargs) -> float:
+
+        """
+        Objective function, returns smaller number for better results
+        """
+
+        simulated_drawdowns = []
+
+        backtest_duration_years = ((max_date-min_date).days/365.2425)
+        trade_count_average_per_year = trade_count/backtest_duration_years
+
+        # add slipage to be closed to live
+        results['profit_percent'] -= SLIPPAGE_PERCENT
```

Comment:
I think you should make here a copy of the dataframe passed, since it not only changes the values in the manner needed for later calculations, needed for internals of this function, but also changes this column in the original dataframe, which will be also used in upper levels of hyperopt (printed as results, etc.). 

If I'm not wrong...

Reply:
`format_results()` is called before this - after `calculate_loss()` the results dataframe is not reused anymore at the moment, so i think this is fine for now (unless we change the order within `generate_optimizer()` ... which i don't think is planned).